src/main/java/io/cdap/plugin/gcp/gcs/StorageClient.java

```diff
@@ -115,9 +115,67 @@ public void mapMetaDataForAllBlobs(String path, Consumer<Map<String, String>> fu
    * @param recursive whether to copy objects in all subdirectories
    * @param overwrite whether to overwrite existing objects
    * @throws IllegalArgumentException if overwrite is false and copying would overwrite an existing object
+   * @return copied blob object from google bucket storage or null if we are dealing with a directory/bucket
    */
-  public void copy(GCSPath sourcePath, GCSPath destPath, boolean recursive, boolean overwrite) {
+  public Blob copy(GCSPath sourcePath, GCSPath destPath, boolean recursive, boolean overwrite) {
+    //If sourcePath is not a bucket and sourcePath doesn't end with "/"
```

Comment:
nit: add space after //

Reply:
> Can you please list the tests that you ran to verify this fix?

Here you have a spreadsheet with all the test cases we've done to verify this fix: 

https://docs.google.com/spreadsheets/d/10CSYkzomHJ3wNiqlqnK7U2y4YxV7PXo3sNhhjx858qI/edit?usp=sharing